### PR TITLE
Fix images and add markdown linter

### PR DESCRIPTION
**Pull request by keturiosakys**
This update pushes some last few fixes for the images before the launch and adds a markdown linter to ensure consistent style in the `docs/` directory.

TBD: custom rules for ReadMe-specific requirements (front matter)

Fixes:
- FP-3003
- FP-3004
- FP-3006

- fix images
- fix code cells
- add markdown linter and fix linter errors
- cleanup
- add github action for linter to check on PRs

### CODE REVIEW

File: docs/providers/tutorial-create-a-provider.md [hunk 1]
```diff
@@ -214,7 +214,7 @@ Try to compile your provider, make sure that it's targeting web assembly and the
 cargo build && wasm-opt -Oz -c -o "./catnip.wasm" "target/wasm32-unknown-unknown/debug/catnip_provider.wasm"
 ```
 
-**Compilation Error**
+##### CompilationError*
```
Comment (arendjr) [inline docs/providers/tutorial-create-a-provider.md:217]:
```suggestion
##### CompilationError
```

**Review by arendjr — APPROVED** [review]

File: docs/providers/tutorial-create-a-provider-advanced.md [hunk 2]
```diff
@@ -1,11 +1,9 @@
 ---
-title: Create a provider -- Advanced topics
+title: Create a provider - advanced
 category: 63d7e8bdbf7b4b0e0745e823
 slug: tutorial-create-a-provider-advanced
 ---
 
-# Create a provider (advanced topics)
-
 This builds on the [Creating a provider tutorial](doc:creating-a-provider), and touches additional topics
```
Comment (gagbo) [inline docs/providers/tutorial-create-a-provider-advanced.md:7]:
```suggestion
This builds on the [Creating a provider tutorial](doc:create-a-provider), and touches additional topics
```

I forgot to update the link

File: docs/providers/tutorial-create-a-provider.md (hunk 1 above)
Comment (gagbo) [inline docs/providers/tutorial-create-a-provider.md:217]:
```suggestion
##### CompilationError
```

File: docs/providers/tutorial-create-a-provider.md [hunk 3]
```diff
@@ -231,7 +231,7 @@ And try to compile again with `cargo build`.
 
 If it still fails, then it's an issue with our own provider development kit. Please file an issue to the repo with the title "Sample Provider does not compile".
 
-**Wasm optimization error**
+##### Wasmoptimizationerror
```
Comment (gagbo) [inline docs/providers/tutorial-create-a-provider.md:234]:
```suggestion
##### Wasm optimization error
```

File: docs/quickstart/Deploy to Docker.md [hunk 4]
```diff
@@ -8,7 +8,7 @@ slug: deploy-to-docker
 
 ### Generate a `fpd` API Token in the Studio
 
-![Register a proxy](quickstart/register_a_proxy.png)
+![Register a proxy](/docs/quickstart/register_an_fpd.png)
```
Comment (gagbo) [inline docs/quickstart/Deploy to Docker.md:11]:
```suggestion
![Register a daemon](/docs/quickstart/register_an_fpd.png)
```

**Review by gagbo — APPROVED** [review]
Few typos here and there, and things I forgot to update in my initial PR, but good to go

**Pull request merged**